2020/day-2-password-philosophy/index.ts

```diff
@@ -0,0 +1,33 @@
+const parseList = (list: string) => {
+  return list
+    .trim()
+    .split('\n')
+    .map(entry => entry.split(': '))
+    .map(([policy, password]) => {
+      const [range, char] = policy.trim().split(' ')
+      const [min, max] = range.split('-').map(Number)
+
+      return { min, max, char, password }
+    })
+}
+
+export const getValidPasswords1 = (list: string) => {
+  const entries = parseList(list)
```

Comment:
I wonder if it makes sense to put this in a const outside of the methods so we don't need to duplicate it?

Reply:
Hm, but we need to pass `list` to `parseList`, which is currently only available as a parameter of these `getValidPasswords*` functions